src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -7498,6 +7498,8 @@ components:
       allOf:
```

Comment:
Do we want to make criteria a required property on CriteriaGroup? That seems most consistent with the approach being used here. Otherwise, we would need to handle the case where criteria is null on criteria group

Reply:
Yes, I'm going to make this into a separate ticket and mob about what we think should be required.

Reply:
I only wanted to change the different types of `SnapshotBuilderCriteria` because I think that's still in the scope of this PR.